packages/@webex/plugin-meetings/src/meeting/voicea-meeting.ts

```diff
@@ -0,0 +1,187 @@
+export const getSpeaker = (members, csis = []) =>
+  Object.values(members).find((member) => {
+    const memberCSIs = member.participant.status.csis ?? [];
+
+    return csis.some((csi) => memberCSIs.includes(csi));
+  });
+
+export const getSpeakerFromProxyOrStore = ({csisKey, meetingMembers, transcriptData}) => {
+  let speaker = {
+    speakerId: '',
+    name: '',
+  };
+
+  let needsCaching = false;
+
+  if (csisKey && transcriptData.speakerProxy[csisKey]) {
+    speaker = transcriptData.speakerProxy[csisKey];
+  }
+  const meetingMember = getSpeaker(meetingMembers, [csisKey]);
+
+  const speakerInStore = {
+    speakerId: meetingMember?.participant.person.id ?? '',
+    name: meetingMember?.participant.person.name ?? '',
+  };
+
+  if (
+    meetingMember &&
+    (speakerInStore.speakerId !== speaker.speakerId || speakerInStore.name !== speaker.name)
+  ) {
+    needsCaching = true;
+    speaker = speakerInStore;
+  }
+
+  return {speaker, needsCaching};
+};
+export const processNewCaptions = ({data, meeting}) => {
+  // TODO: processing the members
```

Comment:
As per the discussion in space, Web Client tested with 1k users and it worked fine. Can close this Todo for now